api/v1alpha7/types.go

```diff
@@ -163,6 +163,24 @@ type RootVolume struct {
 	AvailabilityZone string `json:"availabilityZone,omitempty"`
 }
 
+type DataVolume struct {
+	// NameSuffix is the suffix to be added to volume names.
+	// It will be combined with the machine name to make the volume name.
+	NameSuffix string `json:"nameSuffix"`
+	// Size is the size in GB of the volume.
+	Size int `json:"diskSize"`
+	// VolumeType is the volume type of the volume.
+	// If omitted, the default type will be used.
+	VolumeType string `json:"volumeType,omitempty"`
+	// AvailabilityZone is the volume availability zone to create the volume in.
+	// If omitted, the availability zone of the server will be used.
+	AvailabilityZone string `json:"availabilityZone,omitempty"`
+	// Tag is a tag to add to the volume attachment.
+	// This is then available in the instance metadata, allowing devices to be
+	// identified by tag inside the server.
+	Tag string `json:"tag,omitempty"`
+}
```

Comment:
This can be addressed in a separate PR but for supporting local disks, we'll need to add more fields.

Reply:
Just one I think, to indicate whether it should be a local or Cinder disk? The rest can be reused in both cases.

Reply:
*If* we accept my argument that there will be a single local disk then I would prefer it to have a separate field because it's more obvious to the user and doesn't require validation. I would only put a local flag here if we decide it's useful to have multiple local disks.

But as you say we don't need to consider this yet.

Reply:
@mdbooth as I raised in Slack, I think I have a use case for multiple ephemeral devices carved out of the ephemeral volume, so I would like to see an option that permits that.

Reply:
But it is something for a future PR anyway

Reply:
@mkjpryor Please look at my recent refactor & patches, I think the way we're proposing now is the same behaviour as with rootVolumes. And indeed this can be bulked changed later.